utilities/ldap_sync.sql

```diff
@@ -25,118 +152,763 @@ Changes in this version:
+    # Ensure Connection String has a proper LDAP server
+    #----------------------------------------
+    try:
+        uri = exa.get_connection(ctx.LDAP_CONNECTION).address
+        ldap_server = uri
+    except Exception as e:
+        raise Exception("Unable to find/parse exa.get_connection(ctx.LDAP_CONNECTION).address", exa.meta.script_name, (str(e), abortMessage))   
+
+    server, port = extract_host(ldap_server)
+    
+    #----------------------------------------
+    # Validate just the port extraced from the LDAP CONNECTION
+    #----------------------------------------
+    try:
+        if isinstance(port, int) and (port > 0):
+            pass
+        else:
+            raise Exception(f"Error, trying to validate a numeric Port NUMBER returned: {port} -- from LDAP CONNECTION entry", ldap_server)
+    except Exception as e:
+        raise Exception("Error, unable to extract valid values for the PORT as found in the LDAP Entry:", ldap_server, (str(e), abortMessage))
+    
+          
+    #----------------------------------------
+    # Start building a socket to then test the Host and port for reachability
+    #----------------------------------------
+    try:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    except Exception as e:
+        raise Exception("Unable to open socket", exa.meta.script_name, (str(e), abortMessage))
+    
+    #----------------------------------------
+    # Set socket to timeout after 30 seconds
+    #----------------------------------------
+    try:
+        sock.settimeout(30)
+    except Exception as e:
+        raise Exception("Unable to set socket timeout", exa.meta.script_name, (str(e), abortMessage))
+
+    #----------------------------------------
+    # Make socket connection to LDAP Host and port
+    #----------------------------------------
+    try:
+        global result
+        result = sock.connect_ex((server, port))
+    except Exception as e:
+        if result == 11:
+            raise Exception("Socket timeout...unable to connect to " + str(server) + " and port " + str(port), exa.meta.script_name, (str(e), abortMessage))
+        else:
+            raise Exception("Can open socket, but unable to connect to " + str(server) + " and port " + str(port), exa.meta.script_name, (str(e), abortMessage))
+    
+    #----------------------------------------
+    # If result is anything but 0, then the port was not available
+    #----------------------------------------
+    try:
+        if int(result):
+            raise Exception("Either the host " + str(server) + " is not reachable or Port " + str(port) + " is not reachable on " + server, exa.meta.script_name)
+    except Exception as e:
+        raise Exception("From socket connection on host " + server + " port " + str(port) + " did not work", exa.meta.script_name, (str(e), abortMessage))
+        
+    #========================================
+    # House Keeping - Validate remaining Connection properties
```

Comment:
I see all of these checks multiple times in different scripts. You could put them in one script and use exa.import_script to use the code in multiple scripts.